### DISCUSSION TITLE
map another key to a map overridden by mini

### DISCUSSION THREAD

**Discussion by beaumccartney** [nvim-mini/mini.nvim#1009] [Q&A] [answered]
How can I map something to a default keybind overridden by mini?

In my case, I want to use `gx`, but on a different key. Mini operators uses `gx` as the prefix for exchange, and remapping just that key messes up my muscle memory (many mini keymaps use g<something> as a prefix).

**Comment by echasnovski** [comment, marked as answer]
Unfortunately, there is no easy answer to that.

The most robust way is to use some version of the following code (here it is for your use case):

```lua
local remap = function(mode, lhs_from, lhs_to)
  local keymap = vim.fn.maparg(lhs_from, mode, false, true)
  local rhs = keymap.callback or keymap.rhs
  if rhs == nil then error('Could not remap from ' .. lhs_from .. ' to ' .. lhs_to) end
  vim.keymap.set(mode, lhs_to, rhs, { desc = keymap.desc })
end
remap('n', 'gx', '<Leader>gx')
remap('x', 'gx', '<Leader>gx')
```

**Reply by beaumccartney** [reply]
thanks a ton!
